build/evergreen_config_lib/testgcpkms.py

```diff
@@ -0,0 +1,120 @@
+
+from collections import OrderedDict as OD
+
+from evergreen_config_generator.functions import (shell_exec, func)
+from evergreen_config_generator.tasks import (NamedTask)
+from evergreen_config_generator.variants import (Variant)
+from evergreen_config_generator.taskgroups import (TaskGroup)
+
+def _create_tasks():
+    passtask = NamedTask (task_name="testgcpkms-task")
+
+    passtask.commands = [
+        func("fetch source"),
+        shell_exec (r'''
+            echo "Building test-gcpkms ... begin"
+            pushd mongoc
+            ./.evergreen/compile-test-gcpkms.sh
+            popd
+            echo "Building test-gcpkms ... end"
+            echo "Copying files ... begin"
+            export GCPKMS_GCLOUD=${GCPKMS_GCLOUD}
+            export GCPKMS_PROJECT=${GCPKMS_PROJECT}
+            export GCPKMS_ZONE=${GCPKMS_ZONE}
+            export GCPKMS_INSTANCENAME=${GCPKMS_INSTANCENAME}
+            DRIVERS_TOOLS=$(pwd)/drivers-evergreen-tools
+            mkdir testgcpkms
+            cp ./mongoc/src/libmongoc/test-gcpkms ./mongoc/install/lib/libmongocrypt.* testgcpkms
+            tar czf testgcpkms.tgz testgcpkms/*
+            GCPKMS_SRC="testgcpkms.tgz" GCPKMS_DST=$GCPKMS_INSTANCENAME: $DRIVERS_TOOLS/.evergreen/csfle/gcpkms/copy-file.sh
+            echo "Copying files ... end"
+            echo "Untarring file ... begin"
+            GCPKMS_CMD="tar xf testgcpkms.tgz" $DRIVERS_TOOLS/.evergreen/csfle/gcpkms/run-command.sh
+            echo "Untarring file ... end"
+            ''', test=False),
+           shell_exec (r'''
+            export GCPKMS_GCLOUD=${GCPKMS_GCLOUD}
+            export GCPKMS_PROJECT=${GCPKMS_PROJECT}
+            export GCPKMS_ZONE=${GCPKMS_ZONE}
+            export GCPKMS_INSTANCENAME=${GCPKMS_INSTANCENAME}
+            DRIVERS_TOOLS=$(pwd)/drivers-evergreen-tools
+            GCPKMS_CMD="LD_LIBRARY_PATH=./testgcpkms MONGODB_URI='mongodb://localhost:27017' ./testgcpkms/test-gcpkms" $DRIVERS_TOOLS/.evergreen/csfle/gcpkms/run-command.sh
+            ''')]
+
+    failtask = NamedTask(task_name="testgcpkms-fail-task")
+    failtask.commands = [
+        func("fetch source"),
+        shell_exec (r'''
+            pushd mongoc
+            ./.evergreen/compile-test-gcpkms.sh
+            popd''', test=False),
+         shell_exec (r'''
+            export GCPKMS_GCLOUD=${GCPKMS_GCLOUD}
+            export GCPKMS_PROJECT=${GCPKMS_PROJECT}
+            export GCPKMS_ZONE=${GCPKMS_ZONE}
+            export GCPKMS_INSTANCENAME=${GCPKMS_INSTANCENAME}
+            LD_LIBRARY_PATH=$(pwd)/install MONGODB_URI='mongodb://localhost:27017' EXPECT_ERROR='Failed to connect to: metadata.google.internal' ./mongoc/src/libmongoc/test-gcpkms''')]
+    
+    return [passtask, failtask]
+
+def _create_variant():
+        return Variant(
+        name="testgcpkms-variant",
+        display_name="GCP KMS",
+        # GCP Virtual Machine created is Debian 10.
+        run_on="debian10-small", tasks=[
```

Comment:
I made it debian 10 because I couldn't download libmongocrypt using debian 11 on my local computer since the package wasn't available yet. Is it fine though for the GCP virtual machine to be on debian 11?

Reply:
The GCP virtual machine is already using Debian 11. Suggested change is to have the OS in Evergreen match the GCP OS. libmongocrypt is built from source in Evergreen, so I expect it should be OK.